src/cs1/stacks-and-queues/stack-basics.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Stack Basics
+author:
+  - Jacob Gadberry
+---
+
+# Introduction
+
+The next data structure that we will study is called a stack. The stack data structure just represents a stack of things, either primitives or structs that you define. We will learn about how implement stacks and a few of their applications.
+
+# What is a Stack?
+
+According to Don Knuth, a stack is "a linear list for which all insertions and deletions are made at one end of the list." The quote illustrates the `Last in First Out` principle (LIFO) which all stacks follow. The last item placed in the stack will be the first item to be removed.
```

Comment:
Maybe a "real world" example of a stack might be helpful? For instance I've always liked the analogy to a "stack" of paper, e.g. "pushing" to a stack is like adding another paper to your "stack" of paper, "popping" is removing it from the top, etc.

Reply:
That's a good idea. Wasn't sure which example to use, but paper or plates should work.